src/websocket_account_subscriber.rs

```diff
@@ -0,0 +1,138 @@
+            pubkey,
+            commitment,
+            subscribed: false,
+            event_emitter,
+            unsubscriber: None,
+        }
+    }
+
+    pub async fn subscribe(&mut self) -> SdkResult<()> {
+        if self.subscribed {
+            return Ok(());
+        }
+
+        self.subscribed = true;
+        self.subscribe_ws().await?;
+        Ok(())
+    }
+
+    async fn subscribe_ws(&mut self) -> SdkResult<()> {
+        let account_config = RpcAccountInfoConfig {
+            commitment: Some(self.commitment),
+            encoding: Some(UiAccountEncoding::Base64),
+            ..RpcAccountInfoConfig::default()
+        };
+
+        let pubsub = PubsubClient::new(&self.url).await?;
+        let (unsub_tx, mut unsub_rx) = tokio::sync::mpsc::channel::<()>(1);
+        self.unsubscriber = Some(unsub_tx);
+
+        tokio::spawn({
+            let event_emitter = self.event_emitter.clone();
+            let mut latest_slot = 0;
+            let susbscription_name = self.subscription_name;
+            let pubkey = self.pubkey.clone();
+            async move {
+                let (mut account_updates, account_unsubscribe) = pubsub
+                    .account_subscribe(&pubkey, Some(account_config))
+                    .await
+                    .unwrap();
+
+                loop {
+                    tokio::select! {
+                        message = account_updates.next() => {
+                            match message {
+                                Some(message) => {
+                                    let slot = message.context.slot;
+                                    if slot >= latest_slot {
+                                        latest_slot = slot;
+                                        let account_update = AccountUpdate {
+                                            pubkey: pubkey.to_string(),
+                                            data: message.value,
+                                            slot,
+                                        };
+                                        event_emitter.emit(susbscription_name, Box::new(account_update));
+                                    }
+                                }
+                                None => {
+                                    log::warn!("{}: Account stream interrupted", susbscription_name);
+                                    account_unsubscribe().await;
+                                    break;
+                                }
+                            }
+                        }
+                        unsub = unsub_rx.recv() => {
+                            if let Some(_) = unsub {
+                                log::debug!("{}: Unsubscribing from account stream", susbscription_name);
+                                account_unsubscribe().await;
+                                break;
+
+                            }
+                        }
+                    }
+                }
+            }
```

Comment:
we will need some way to indicate when this subscription task has ended or auto-reconnection, otherwise will get silent failures